cpp/src/arrow/compute/kernels/scalar_string_test.cc

```diff
@@ -712,6 +712,95 @@ TEST_F(TestFixedSizeBinaryKernels, BinaryLength) {
              "[6, null, 6]");
 }
 
+TEST_F(TestFixedSizeBinaryKernels, SliceBytesBasic) {
+  SliceOptions options{2, 4};
+  CheckUnary("binary_slice", R"(["abcabc", "defdef"])", fixed_size_binary(2),
```

Comment:
Can we use less repeated values to make sure the substring selection is right?
```suggestion
  CheckUnary("binary_slice", R"(["abcdef", "ghijkl"])", fixed_size_binary(2),
```

Reply:
Also, can you add some null values at some point?

Reply:
Done.